### DISCUSSION TITLE
Training functionality

### DISCUSSION THREAD

**Discussion by jatkinson1000** [Cambridge-ICCS/FTorch#22] [Ideas]
As a first instance (and likely first release) we only implemented inference functionalities from Torch as this was a good starting point. The assumption was that ML developers would build and train their models from within python, only porting to Fortran (or C?) for deployment.

However, within our userspace it is becoming apparent that there may well be a significant use-case for online training, especially to aid with model-larger code interfacing and preventing blowup etc.

Do we want to look into adding this capability in a future release, and what would be our guidelines for deploying (e.g. build in Python)?

**Comment by dorchard** [comment]
Thanks for starting this discussion. I think we see how things go with adoption first and then see whether it makes sense putting in the effort to add this.

**Comment by siddanib** [comment]
Thanks for making this library publicly available! This might be something that you would have already taken into account. I hope the online training considers distributed memory simulations, and therefore, distributed training of the ML models.
